packages/app/src/cli/models/extensions/schemas.ts

```diff
@@ -32,26 +32,27 @@ export const ApiVersionSchema = zod.string()
 
 export type ApiVersionSchemaType = zod.infer<typeof ApiVersionSchema>
 
-export const BaseUIExtensionSchema = zod.object({
+export const BaseSchema = zod.object({
   name: zod.string(),
+  type: zod.string(),
   description: zod.string().optional(),
-  type: zod.string().default('ui_extension'),
   apiVersion: ApiVersionSchema.optional(),
   extensionPoints: zod.any().optional(),
   capabilities: CapabilitiesSchema.optional(),
   metafields: zod.array(MetafieldSchema).optional().default([]),
   categories: zod.array(zod.string()).optional(),
 })
 
-export const ThemeExtensionSchema = zod.object({
+export const BaseUIExtensionSchema = BaseSchema.extend({
+  type: zod.string().default('ui_extension'),
+})
+
+export const ThemeExtensionSchema = BaseSchema.extend({
   name: zod.string(),
   type: zod.literal('theme'),
 })
 
-export const BaseFunctionConfigurationSchema = zod.object({
-  name: zod.string(),
-  type: zod.string(),
-  description: zod.string().optional().default(''),
+export const BaseFunctionConfigurationSchema = BaseSchema.extend({
```

Comment:
nitpick, why these names?
- BaseUIExtensionSchema -> UIExtensionSchema
- BaseFunctionConfigurationSchema -> FunctionSchema

or am I missing something?

Reply:
nop, you are right, I messed the names here during the refactor

Reply:
Is because Functions used to have two schemas: one for configuration and one for metadata (that no longer exists)